### PR TITLE
Use `like_regex` operator for regex filters

### PR DESCRIPTION
**Pull request by galibey**
Change regex filter format from `... ~= /PATTERN/FLAGS` to SQL/JSON one `... like_regex "PATTERN"`. Special handling for pattern flags removed. They can be specified inside the pattern.

### CODE REVIEW

File: jsurfer-core/src/main/java/org/jsfr/json/filter/MatchRegexPredicate.java [hunk 1]
```diff
@@ -46,10 +46,16 @@ public MatchRegexPredicate(JsonPath relativePath, Pattern regex) {
...
             Object candidate = primitiveHolder.getValue();
-            String string = (String) jsonProvider.cast(candidate, String.class);
-            return regex.matcher(string).find();
+            return regex.matcher(cast(candidate, jsonProvider)).find();
         }
         return false;
     }
 
+    private static String cast(Object candidate, JsonProvider jsonProvider) {
+        if (candidate instanceof Number || candidate instanceof String) {
```
Comment (viliam-durina) [inline jsurfer-core/src/main/java/org/jsfr/json/filter/MatchRegexPredicate.java]: Why wasn't `cast(candidate, jsonProvider)` sufficient?

Comment (viliam-durina) [reply jsurfer-core/src/main/java/org/jsfr/json/filter/MatchRegexPredicate.java]: Tried it and saw which tests fail. But I'd say the code should remain here. The issue is in the provider's `cast` method.

Comment (galibey) [reply jsurfer-core/src/main/java/org/jsfr/json/filter/MatchRegexPredicate.java]: Yes, providers should have handled that but some, including the one we use in `hazelcast-sql` `JacksonJrProvider`, don't, so we have to work around it for basic types that we know how to convert to String.

Comment (galibey) [reply jsurfer-core/src/main/java/org/jsfr/json/filter/MatchRegexPredicate.java]: Boolean also should be there

Comment (viliam-durina) [reply jsurfer-core/src/main/java/org/jsfr/json/filter/MatchRegexPredicate.java]: But shouldn't we fix the providers?

Comment (galibey) [reply jsurfer-core/src/main/java/org/jsfr/json/filter/MatchRegexPredicate.java]: You're right. For some reason, I thought they are external dependency for us. Will do it.

**Review by viliam-durina — APPROVED** [review]

**Pull request merged**